giskard_vision/core/scanner/scanner.py

```diff
@@ -129,11 +165,12 @@ def get_detectors(self, tags: Optional[Sequence[str]] = None) -> Sequence:
         detectors = []
         classes = DetectorRegistry.get_detector_classes(tags=tags)
 
-        # Filter detector classes
-        if self.only:
-            only_classes = DetectorRegistry.get_detector_classes(tags=self.only)
-            keys_to_keep = set(only_classes.keys()).intersection(classes.keys())
-            classes = {k: classes[k] for k in keys_to_keep}
+        # Filter vision detector classes
+        print(classes)
+        vision_classes = DetectorRegistry.get_detector_classes(tags=["vision"])
+        only_classes = DetectorRegistry.get_detector_classes(tags=self.only if self.only else None)
```

Comment:
I don't get it, why not do:
```py
classes = DetectorRegistry.get_detector_classes(tags=["vision"]+ self.only if self.only else [])
```

Reply:
Cause we want to get the intersection between vision and only, not the union, and I think this computes the union

Reply:
ok but can you factor out the self.only outside? i.e. do not run `get_detector_classes` if self.only is None